plugins/inputs/cloud_pubsub/cloud_pubsub.go

```diff
@@ -177,20 +177,22 @@ func (ps *PubSub) onMessage(ctx context.Context, msg message) error {
 	}
 
 	var data []byte
-	if ps.Base64Data {
-		strData, err := base64.StdEncoding.DecodeString(string(msg.Data()))
-		if err != nil {
-			return fmt.Errorf("unable to base64 decode message: %w", err)
-		}
-		data = strData
-	} else {
-		data = msg.Data()
+	var err error
+
+	data, err = ps.decompressData(msg.Data())
```

Comment:
Use
```suggestion
	data, err := ps.decompressData(msg.Data())
```
and also remove the `var data []byte` statement in line 179. This will make the code more compact.